pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -266,6 +273,34 @@ func (r *components) getComponent(ctx context.Context, acc v1alpha2.ApplicationC
 	return c, revisionName, nil
 }
 
+// pass through labels and annotation from app-config to workload  or trait
+func (r *components) passThroughObjMeta(oMeta metav1.ObjectMeta,u *unstructured.Unstructured) error{
+	// pass app-config labels
+	curLabels := oMeta.GetLabels()
+	if curLabels != nil && len(curLabels) > 0 {
+		// workload or trait labels
+		dstLabels := u.GetLabels()
+		if dstLabels == nil {
+			dstLabels = map[string]string{}
+		}
+		mergo.Map(&dstLabels,curLabels)
```

Comment:
what does this do? We kinda out-source the entire core logic of this PR to this library.

Reply:
Merging maps is a simple logic, encapsulates a function to replace external library dependencies